src/System.Diagnostics.Process/tests/ProcessTestBase.cs

```diff
@@ -51,14 +51,20 @@ protected void AddProcessForDispose(Process p)
 
         protected Process CreateProcess(Func<int> method = null)
         {
-            Process p = RemoteInvoke(method ?? (() => SuccessExitCode), new RemoteInvokeOptions { Start = false }).Process;
+            RemoteInvokeHandle handle = RemoteInvoke(method ?? (() => SuccessExitCode), new RemoteInvokeOptions { Start = false });
+            Process p = handle.Process;
+            handle.Process = null;
+            handle.Dispose();
             AddProcessForDispose(p);
```

Comment:
another option could be to pass the handle to AddProcessFordispose instead of the process itself. then you could remove 55-57 in both cases.